src/java.base/share/classes/com/sun/crypto/provider/PKCS12PBECipherCore.java

```diff
@@ -287,249 +308,170 @@ void implInit(int opmode, Key key, AlgorithmParameterSpec params,
                 throw new InvalidAlgorithmParameterException
                         ("IterationCount must be a positive number");
             }
+
             byte[] derivedKey = derive(passwdChars, salt, iCount,
                     keySize, CIPHER_KEY);
             SecretKey cipherKey = new SecretKeySpec(derivedKey, algo);
             Arrays.fill(derivedKey, (byte)0);
 
-            try {
-                if (cipherImpl instanceof ARCFOURCipher rc4) {
-                    rc4.engineInit(opmode, cipherKey, random);
-                } else {
+            KeyAndIv result;
```

Comment:
I think the fancy 2022 way is to `return switch (...) { case "RC4" -> new KeyAndIv(...) ... }`.

Reply:
Ok~